cpp/tests/filter/filter_test.cpp

```diff
@@ -0,0 +1,299 @@
+/*
+ * Copyright (c) 2025, NVIDIA CORPORATION.
+
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cudf_test/base_fixture.hpp>
+#include <cudf_test/column_wrapper.hpp>
+#include <cudf_test/random.hpp>
+#include <cudf_test/type_lists.hpp>
+
+#include <cudf/detail/iterator.cuh>
+#include <cudf/filter.hpp>
+#include <cudf/jit/runtime_support.hpp>
+
+namespace filters {
+
+struct TypeSupportTest : public cudf::test::BaseFixture {
```

Comment:
Could we name these differently to tell them apart from other tests perhaps?
```suggestion
struct FilterTestFixture: public cudf::test::BaseFixture {
```